src/main/java/de/retest/recheck/util/OptionalUtil.java

```diff
@@ -0,0 +1,22 @@
+package de.retest.recheck.util;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+public class OptionalUtil {
+
+	private OptionalUtil() {}
+
+	@SuppressWarnings( "unchecked" )
+	public static <T> Stream<T> stream( final Optional<? extends T> o ) {
+		return (Stream<T>) o.map( Stream::of ).orElseGet( Stream::empty );
+	}
```

Comment:
Do we want to use a real method and not a lambda?

```java
private static final Function<Optional<?>, Stream<?>> toStream = o -> o.map( Stream::of ).orElse( Stream.empty() );

public static <T> Function<? super Optional<T>, Stream<T> stream() {
    return /*Fancy cast here*/ toStream;
}
```

Reply:
Dunno, what is the benefit?

Reply:
Maybe increased readability on site of the usage and&mdash;on a lower note&mdash;fewer objects created?

```java
list.stream() //
    .map( OptionalUtil::stream )

list.stream() //
    .map( OptionalUtil.stream() )
```

But same as the discussion about complex lambda vs method reference within a static method: No real preference.

Reply:
I think I personally prefer the current, method-ref-based style. But thanks for bringing attention to this.